### PR TITLE
fix(mjs): remove harmony unused build errors

### LINKED ISSUE #9044: .mjs ESM wildcard import from .js CJS causes incorrect build error
# Bug report

**What is the current behavior?**

An ESM wildcard import from a CJS `.js` file within a `.mjs` file incorrectly causes the following Webpack build error:

```
Can't import the namespace object from non EcmaScript module (only default export is available)
```

In reality ESM wildcard imports from CJS files has always worked in Node.js `--experimental-modules` mode:

Screen Shot 2019-04-17 at 5 32 13 pm

**If the current behavior is a bug, please provide the steps to reproduce.**

With `a.js`:

```js
module.exports = {
  foo: 'bar'
}
```

And `b.mjs`:

```js
import * as a from './a'
console.log(a)
```

Run:

```sh
webpack b.mjs
```

To see the incorrect build error:

Screen Shot 2019-04-17 at 5 55 50 pm

**What is the expected behavior?**

There should not be a build error.

If I change this line in the Webpack code:

https://github.com/webpack/webpack/blob/v4.30.0/lib/dependencies/HarmonyImportSpecifierDependency.js#L83

To:

```diff
- this._id !== "default"
+ this._id !== "default" && this._id !== null
```

The problem goes away. Although I don't know what the proper way to detect a wildcard import is.

**Other relevant information:**
webpack version:  4.30.0
Node.js version: 8.5.0 - 11.14.0
Operating System: macOS Mojave
Additional tools: N/A

### PR DESCRIPTION
**Pull request by alreadyExisted**
**What kind of change does this PR introduce?**

fixes #9044

**Did you add tests for your changes?**

No

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

nothing

### CODE REVIEW

**Comment by alreadyExisted** [comment]
I hope I did not delete too much

File: lib/dependencies/HarmonyImportSpecifierDependency.js [hunk 1]
```diff
@@ -77,21 +77,6 @@ class HarmonyImportSpecifierDependency extends HarmonyImportDependency {
 		}
 
 		if (!importedModule.buildMeta || !importedModule.buildMeta.exportsType) {
-			// It's not an harmony module
```
Comment (sokra) [inline lib/dependencies/HarmonyImportSpecifierDependency.js]: Same

File: test/cases/mjs/cjs-import-default/errors.js [hunk 2]
```diff
@@ -1,20 +0,0 @@
-module.exports = [
-	[
-		/Can't import the namespace object from non EcmaScript module \(only default export is available\)/
-	],
-	[
-		/Can't import the namespace object from non EcmaScript module \(only default export is available\)/
-	],
-	[
```
Comment (sokra) [inline test/cases/mjs/cjs-import-default/errors.js]: 3 / 6 errors should stay.

File: lib/dependencies/HarmonyExportImportedSpecifierDependency.js [hunk 3]
```diff
@@ -353,9 +353,7 @@ class HarmonyExportImportedSpecifierDependency extends HarmonyImportDependency {
 				this.id !== "default"
```
Comment (sokra) [inline lib/dependencies/HarmonyExportImportedSpecifierDependency.js]:
```suggestion
				this.id && this.id !== "default"
```

File: lib/dependencies/HarmonyImportSpecifierDependency.js [hunk 4]
```diff
@@ -83,9 +83,7 @@ class HarmonyImportSpecifierDependency extends HarmonyImportDependency {
 				this._id !== "default"
```
Comment (sokra) [inline lib/dependencies/HarmonyImportSpecifierDependency.js]: Same

**Comment by alreadyExisted** [comment]
Thanks, sorry, hurried

**Comment by sokra** [comment]
For some weird reason this PR seem to be broken somehow...

created new PR: #9260

**Pull request closed without merging**